src/waveform/waveformwidgetfactory.cpp

```diff
@@ -463,7 +491,7 @@ bool WaveformWidgetFactory::setWidgetTypeFromHandle(int handleIndex) {
     }
 
     WaveformWidgetAbstractHandle& handle = m_waveformWidgetHandles[handleIndex];
-    if (handle.m_type == m_type) {
+    if (handle.m_type == m_type && !force) {
```

Comment:
I wonder what code exactly in the key for the fix? 

Is it: 
holder.m_waveformWidget = widget;
viewer->setWaveformWidget(widget);
widget->getWidget()->show();
viewer->update();

If we would know that, we can call this in the beginning of the loop and the patch becomes real small.

Can you check it?

Reply:
Remember that RJ also made some attempts in  #2681 that didn't work here.

Reply:
Ok, so no need to do this again.

Reply:
I've been trying for a while adding the waveformviewer bits of that PR to mine (and commenting my doublecreation) but nothing worked.
I still don't quite get the logic, but with the additional logging on RJ's PR, I see how the exposed flag changes:

When I create the object once (which happens way before these log lines)
```
Debug [Main]: WWaveformViewer(0x2e8c53cec80) resizeEvent
Debug [Main]: WWaveformViewer(0x2e8c53cec80) isVisible() true QSize(1507, 73)
Debug [Main]: WWaveformViewer(0x2e8c53cec80) isVisible() WWidgetGroup(0x2e8ad34ad80, name="Waveform") QSize(1585, 73)
Debug [Main]: WWaveformViewer(0x2e8c53cec80) widget isVisible() true QSize(22, 10)
Debug [Main]: WWaveformViewer(0x2e8c53cec80) widget isVisible() true QSize(1507, 73)
Debug [Main]: QWidgetWindow(0x2e8d0246070, name="QGLWidgetClassWindow") vis true QSize(1507, 73) QSurface::OpenGLSurface exposed false
```
When I do the duplication
```
Debug [Main]: WWaveformViewer(0x1f7559809f0) resizeEvent
Debug [Main]: WWaveformViewer(0x1f7559809f0) isVisible() true QSize(1507, 73)
Debug [Main]: WWaveformViewer(0x1f7559809f0) isVisible() WWidgetGroup(0x1f75f0f3200, name="Waveform") QSize(1585, 73)
Debug [Main]: WWaveformViewer(0x1f7559809f0) widget isVisible() true QSize(22, 10)
Debug [Main]: WWaveformViewer(0x1f7559809f0) widget isVisible() true QSize(1507, 73)
Debug [Main]: QWidgetWindow(0x1f76a601070, name="QWidgetClassWindow") vis true QSize(1507, 73) QSurface::OpenGLSurface exposed true
```

Note: I get 8 resizeEvent with either of both cases. 
Without duplication, all 8 show exposed false.
When I duplicate, the three last show exposed true (which means the exposing happens after the first resizing once the widged is visible), but does not happen without the duplication.

I can share the logs but.. where could I put them?

Reply:
e.g. https://pastebin.com/